controller/contact.js

```diff
@@ -145,11 +143,19 @@ const create = async (req, res, _) => {
 };
 
 const remove = async (req, res, _) => {
+  const { _id: userId } = req.user;
   const { contactId } = req.params;
 
   try {
     const deletedContact = await service.removeContact(contactId);
-    res.json(genereteJSON("success", 200, "contact", deletedContact));
+
+    if (deletedContact.owner.toString() === userId.toString()) {
```

Comment:
Takie coś powinno nie powinno mieć miejsca. Czyli w linii 150 prześlij jeszcze userId razem z contactId. Jeśli to zostanie wykonane to wtedy wiesz że należało to do tego użytkownika